### PR TITLE
cis aws 4.1 .. 4.15

### PR DESCRIPTION
**Pull request by olegsu**
### Summary of your changes
Add AWS CIS monitoring rules 4.1 ... 4.15

### Related Issues
- Related https://github.com/elastic/csp-security-policies/issues/148
- Related https://github.com/elastic/cloudbeat/pull/668

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: bundle/compliance/policy/aws_cloudtrail/ensure_unauthorized_api_logged.rego [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	data_adapter.is_monitoring_type
+
+	# set result
+	result := common.generate_result_without_expected(
+		common.calculate_result(rule_evaluation),
+		input.resource,
+	)
+}
+
+rule_evaluation {
```
[the remark this answers is not in the document]
Comment (oren-zohar) [reply bundle/compliance/policy/aws_cloudtrail/ensure_unauthorized_api_logged.rego:19]: many of the audit steps are missing, are they explicitly happening in cloudbeat?

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: bundle/compliance/policy/aws_cloudtrail/ensure_unauthorized_api_logged.rego [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	# At least one trail
+	count(data_adapter.trail_items) > 0
```
[the remark this answers is not in the document]
Comment (olegsu) [reply bundle/compliance/policy/aws_cloudtrail/ensure_unauthorized_api_logged.rego:21]: Right, this rule will be evaluated once per cycle with all the trails.
When there are no trails, it means no trail audit authz.

File: bundle/compliance/policy/aws_cloudtrail/ensure_unauthorized_api_logged.rego [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+	# with IsMultiRegionTrail=true
+	count(data_adapter.multi_region_trails) > 0
```
[the remark this answers is not in the document]
Comment (olegsu) [reply bundle/compliance/policy/aws_cloudtrail/ensure_unauthorized_api_logged.rego:24]: Right, from the CIS 
> Identify Multi region Cloudtrails: Trails with "IsMultiRegionTrail" set to true

File: bundle/compliance/policy/aws_cloudtrail/ensure_unauthorized_api_logged.rego (hunk 1 above)
[the remark this answers is not in the document]
Comment (olegsu) [reply bundle/compliance/policy/aws_cloudtrail/ensure_unauthorized_api_logged.rego:19]: Besides the pattern validation (which is passed to OPA as `items.AuthorizationFilterExists` everything is evaluated here.
> Get a list of all associated metric filters for this <cloudtrail_log_group_name> that you captured in step 1:
aws logs describe-metric-filters --log-group-name "<cloudtrail_log_group_name>"
 Ensure the output from the above command contains the following:
 "filterPattern": "{ ($.errorCode = *UnauthorizedOperation) || ($.errorCode = AccessDenied*) || ($.sourceIPAddress!=delivery.logs.amazonaws.com) || ($.eventName!=HeadBucket) }",

[the remark this answers is not in the document]
Comment (olegsu) [reply bundle/compliance/policy/aws_cloudtrail/ensure_unauthorized_api_logged.rego:19]: > where is the check for
> 
> ```
> • Ensure Identified Multi region CloudTrail is active
> aws cloudtrail get-trail-status --name <Name of a Multi-region CloudTrail>
> ensure IsLogging is set to TRUE
> • Ensure identified Multi-region Cloudtrail captures all Management Events
> aws cloudtrail get-event-selectors --trail-name <"Name" as shown in describe- trails>
> Ensure there is at least one Event Selector for a Trail with IncludeManagementEvents set to true and ReadWriteType set to All
> ```

You are right, missed those

File: bundle/compliance/cis_aws/test_data.rego [hunk 4]
```diff
@@ -149,3 +149,9 @@ generate_security_group(entry) = {
 	"type": "ec2",
 	"subType": "aws-security-group",
 }
+
+generate_monitoring_cis_4_1(items) = {
```
[the remark this answers is not in the document]
Comment (olegsu) [reply bundle/compliance/cis_aws/test_data.rego:153]: I am not sure what is the problem having this as well

**Review by oren-zohar — CHANGES_REQUESTED** [review]
why there are no tests for the rules?

**Comment by olegsu** [comment]
> why there are no tests for the rules?

All the rules 4.1 ... 4.15 are mostly the same. The difference is in the pattern that is matched.
I dont think it will help to run the same tests again. 
It was covered [here](https://github.com/elastic/csp-security-policies/pull/179/files#diff-3608b509657870832a4f6cc83cc6309b75af1257529f5a3b3f5f409f59a31b40R1)

File: bundle/compliance/policy/aws_cloudtrail/pattern_test.rego [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+pattern_never_match = "not_match"
+
+test_pass {
+	at_least_one_metric_exists({"MetricFilters": [filter_1]}, [pattern_1])
+	at_least_one_metric_exists({"MetricFilters": [filter_1, filter_2]}, [pattern_1])
+	at_least_one_metric_exists({"MetricFilters": [filter_1, filter_2]}, [pattern_2])
+	at_least_one_metric_exists({"MetricFilters": [filter_1, filter_2]}, [pattern_never_match, pattern_1])
+}
+
+test_fail {
```
[the remark this answers is not in the document]
Comment (olegsu) [reply bundle/compliance/policy/aws_cloudtrail/pattern_test.rego:20]: fail, will add test to validate it. Thanks

File: bundle/compliance/policy/aws_cloudtrail/pattern.rego [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package compliance.policy.aws_cloudtrail.pattern
+
+# check that a trail has least one metmetric filter pattern that match to at least one pattern
+at_least_one_metric_exists(trail, patterns) {
+	some i
+	filter := trail.MetricFilters[i]
+	some j
```
[the remark this answers is not in the document]
Comment (olegsu) [reply bundle/compliance/policy/aws_cloudtrail/pattern.rego:7]: This is iteration over two arrays, not a nested one. 
Will add the here output to show the indices.

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**